docs/architecture/ADR-004-IBC-allowlist.md

```diff
@@ -0,0 +1,27 @@
+# ADR 004: IBC Allowlist
+
+## Terminology
+
+All terminology is defined in [ICS 024](https://github.com/cosmos/ibc/tree/main/spec/core/ics-024-host-requirements) and some references are from corresponding implementation of [ICS 002](https://github.com/cosmos/ibc/tree/main/spec/core/ics-002-client-semantics) in [ibc-go](https://github.com/cosmos/ibc-go/blob/da1b7e0aaf4b7d466b1a7d1ed4f5d81149ff1d5b/modules/core/02-client)
+
+## Changelog
+
+- 2022-03-03: Initial Commit
+
+## Context
+
+While enabling IBC, we want to connect, and allow messages from selected chains. However, we don't want the ability for an arbitrary entity to create an IBC connection with a zone. This is so that we can keep the state machine as minimal and focused as possible.
```

Comment:
I think we first need to fully understand the overhead in the state machine of IBC and its pricing.